src/components/NcActionButton/NcActionButton.vue

```diff
@@ -269,6 +383,35 @@ export default {
 			type: Boolean,
 			default: false,
 		},
+
+		/**
+		 * The button's behavior, by default the button acts like a normal button with optional toggle button behavior if `modelValue` is `true` or `false`.
+		 * But you can also set to checkbox button behavior with tri-state or radio button like behavior.
+		 * This extends the native HTML button type attribute.
+		 */
+		type: {
+			type: String,
+			default: 'button',
+			validator: (behavior) => ['button', 'checkbox', 'radio', 'reset', 'submit'].includes(behavior),
+		},
+
+		/**
+		 * The buttons state if `type` is 'checkbox' or 'radio' (meaning if it is pressed / selected)
+		 * Either boolean for checkbox and toggle button behavior or `value` for radio behavior
+		 */
+		modelValue: {
+			type: [Boolean, String],
+			default: null,
+		},
```

Comment:
It's said "if `type` is 'checkbox' or 'radio'" but it also changes attributes on other types.

And I think it's worth mentioning that `null` (the default value) in a specific case (in a menu with type checkbox) actually means `mixed`, not `false`.

Do we actually need this `mixed` value? It is not supported by `NcButton`, and there is no UI for this value. If feels a bit complicated to me...

Reply:
I added more documentation on this behavior